platform/security/idp/security-idp-plugin/security-idp-plugin-namequalifier/src/test/groovy/org/codice/ddf/security/idp/plugin/namequalifier/NameQualifierPresignPluginSpec.groovy

```diff
@@ -0,0 +1,182 @@
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.idp.plugin.namequalifier
+
+import ddf.security.samlp.SamlProtocol
+import org.apache.wss4j.common.saml.OpenSAMLUtil
+import org.opensaml.saml.saml2.core.Assertion
+import org.opensaml.saml.saml2.core.AuthnRequest
+import org.opensaml.saml.saml2.core.Issuer
+import org.opensaml.saml.saml2.core.Response
+import org.opensaml.saml.saml2.core.Subject
+import org.opensaml.saml.saml2.core.NameID
+import spock.lang.Specification
+import org.apache.wss4j.common.saml.builder.SAML2Constants
+import org.opensaml.saml.saml2.core.NameIDPolicy
+
+class NameQualifierPresignPluginSpec extends Specification {
+
+    NameQualifierPresignPlugin plugin = new NameQualifierPresignPlugin()
+
+    // general mocks
+    AuthnRequest authNReq = Mock(AuthnRequest)
+    Issuer issuer = Mock(Issuer)
+
+    List<String> spMetadata = new ArrayList<>()
+    Set<SamlProtocol.Binding> bindings = new HashSet<>()
+
+    // set of mocks and concrete objects
+    Response response = Mock(Response)
+
+    List<Assertion> assertions = new ArrayList<>()
+    Assertion assertion = Mock(Assertion)
+
+    Subject subject = Mock(Subject)
+    NameID nameID = Mock(NameID)
+
+    NameIDPolicy nameIdPolicy = Mock(NameIDPolicy)
+
+    String format = new String(SAML2Constants.NAMEID_FORMAT_PERSISTENT)
+
+    def NAME_QUALIFIER = 'https://localhost:8993/services/idp/login'
+
+    def PERSISTENT = 'urn:oasis:names:tc:SAML:2.0:nameid-format:persistent'
+
+    def NEGATIVE_TEST = 'negative test'
+
+    def setupSpec() {
+        OpenSAMLUtil.initSamlEngine()
+    }
+
+    def setup() {
+
+        // set up authNReq
+        authNReq.getIssuer() >> issuer
+
+        // set up issuer
+        response.getIssuer() >> issuer
+
+        // set up subject
+        assertion.getIssuer() >> issuer
+
+        // set up basic response
+        response.assertions >> assertions
+        //assertions.add(assertion)
+
+        // set up the name id policy
+        authNReq.getNameIDPolicy() >> nameIdPolicy;
+
+    }
+
+     def 'test name qualifier when issuer format is not null and is persistent'() {
+
+        setup:
+        response.getIssuer().getFormat() >> format
+        assertions.add(assertion)
+        assertion.getSubject() >> subject
+        assertion.getSubject().getNameID() >> nameID
+        nameID.getFormat() >> PERSISTENT
+
+        when:
+        plugin.processPresign(response, authNReq, spMetadata, bindings)
+
+        then:
+        1 * nameID.getFormat()
+
```

Comment:
❓ Shouldn't this run a second time for the second half of the condition? 
❓ Should this also affirm that `1 * nameID.setNameQualifier(NAME_QUALIFIER)`?